packages/ihe-gateway-sdk/src/models/shared.ts

```diff
@@ -99,11 +99,25 @@ export const xcaGatewaySchema = z.object({
 });
 export type XCAGateway = z.infer<typeof xcaGatewaySchema>;
 
+export const XCPDGatewaySchema = z.object({
+  oid: z.string(),
+  url: z.string(),
+  id: z.string().optional(),
+});
+export type XCPDGateway = z.infer<typeof XCPDGatewaySchema>;
+
+export type XCPDGateways = XCPDGateway[];
+
 export const documentReferenceSchema = z.object({
   homeCommunityId: z.string(),
   docUniqueId: z.string(),
-  urn: z.string(),
   repositoryUniqueId: z.string(),
+  fileName: z.string().optional(),
+  bucketName: z.string().optional(),
+  isNew: z.boolean().optional(),
+  size: z.number().optional(),
```

Comment:
Just to make sure we want `.optional()` and not `.nullish()` - could validate this by checking the request done by the IHE GW.

Reply:
So this logic currently isnt in mirth yet, im going to add it another PR but is going to be a part of the schema. Ill change it to nullish though just to be on the safe side